ecommerce/programs/management/commands/migrate_program_offers_to_commercetools.py

```diff
@@ -0,0 +1,395 @@
+import logging
+import re
+from enum import Enum
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db.models import Q
+from oscar.core.loading import get_model
+from requests.exceptions import HTTPError
+
+from ecommerce.extensions.client import CommercetoolsAPIClient
+from ecommerce.programs.utils import get_all_program_uuids
+
+logger = logging.getLogger(__name__)
+
+
+Benefit = get_model('offer', 'Benefit')
+ConditionalOffer = get_model('offer', 'ConditionalOffer')
+SiteConfiguration = get_model('core', 'SiteConfiguration')
+
+
+CT_ABSOLUTE_DISCOUNT_TYPE = 'absolute'
+CT_PERCENTAGE_DISCOUNT_TYPE = 'relative'
+
+TEN_PERCENT_DISCOUNT_IN_CENTS = 1000
+
+CT_CART_DISCOUNT_TYPE_MAP = {
+    Benefit.FIXED: CT_ABSOLUTE_DISCOUNT_TYPE,
+    Benefit.PERCENTAGE: CT_PERCENTAGE_DISCOUNT_TYPE
+}
+
+
+class ProxyClassDiscountType(Enum):
+    """Enumeration of discount types in the proxy class."""
+
+    PERCENTAGE = "ecommerce.programs.benefits.PercentageDiscountBenefitWithoutRange"
+    ABSOLUTE = "ecommerce.programs.benefits.AbsoluteDiscountBenefitWithoutRange"
+
+
+def _get_highest_sort_order(client: CommercetoolsAPIClient):
+    """
+    Get the highest sort order for cart discounts without discount codes.
+
+    Args:
+        client (CommercetoolsAPIClient): Commercetools API client.
+
+    Returns:
+        float: The highest sort order.
+    """
+    response = client.get_highest_sort_order_for_cart_discount_without_codes()
+
+    if not response:
+        raise CommandError("Failed to get highest sort order for cart discounts without codes. Exiting command.")
+
+    if response['count'] > 0:
+        return float(response['results'][0]['sortOrder'])
+
+    return 0.00000000000001
+
+
+def _create_cart_discount(
+    client: CommercetoolsAPIClient,
+    discount_type: str,
+    discount_value: int,
+    sort_order: float,
+    predicate: str
+):
+    """
+    Create a new cart discount.
+
+    Args:
+        client (CommercetoolsAPIClient): Commercetools API client.
+        discount_type (str): Type of discount (e.g., "relative").
+        discount_value (int): Value of the discount.
+        sort_order (float): Sort order (Rank) for the cart discount.
+        predicate (str): Predicate for the cart discount.
+
+    Returns:
+        Dict: Created cart discount data or None if request fails.
+    """
+    display_discount_value = int(discount_value / 100)
+    response = client.create_cart_discount_without_code(
+        key=f"{discount_type}-{display_discount_value}-program-offer",
```

Comment:
Please make `program-offer` a constant

```suggestion
        key=f"{discount_type}-{display_discount_value}-program-offer",
```